src/autolabel/configs/schema.py

```diff
@@ -0,0 +1,94 @@
+"""Config Schema"""
+
+# Schema definition for Autolabel API/ calls
+# 'additionalProperties' is set to False for al properties to handle typos
+# for not required properties/fields.
+schema = {
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "Label Config",
+    "description": "The query configuration to generate autolabels",
+    "type": "object",
+    "properties": {
+        "task_name": {
+            "type": "string",
+            "description": "The task name of the labeling job",
+        },
+        "task_type": {
+            "enum": [
+                "classification",
+                "entity_matching",
+                "named_entity_recognition",
+                "question_answering",
+            ],
+            "description": "The type of auto labeling task",
+        },
+        "dataset": {
+            "type": "object",
+            "properties": {
+                "label_column": {"type": "string"},
+                "text_column": {"type": "string"},
+                "delimiter": {"type": "string"},
+                "explanation_column": {"type": "string"},
+            },
+            "additionalProperties": False,
+        },
+        "model": {
+            "type": "object",
+            "properties": {
+                "provider": {
+                    "enum": [
+                        # The below are supported model providers
+                        "huggingface_pipeline",
+                        "refuel",
+                        "openai",
+                        "google",
+                        "anthropic",
+                    ],
```

Comment:
can we have this be populated dynamically from https://github.com/refuel-ai/autolabel/blob/main/src/autolabel/schema.py ?